internal/install/recipes/bundler.go

```diff
@@ -80,9 +88,21 @@ func (b *Bundler) getBundleRecipeWithDependencies(recipe *types.OpenInstallation
 		Recipe: recipe,
 	}
 
+	// For OHIs with dual infrastructure-agent/super-agent dependencies, remove the one not needed:
+	//	if super-agent is targeted, infrastructure-agent is removed
+	//	if super-agent is not targeted, super-agent is removed
+	if recipe.IsOhi() && utils.StringInSlice(types.InfraAgentRecipeName, recipe.Dependencies) && utils.StringInSlice(types.SuperAgentRecipeName, recipe.Dependencies) && depException != "" {
```

Comment:
We don't want to hardcode logic based on recipe name (such as super-agent).
We just want to check if there is a target install recipe:
If there is no target install: add/check the first dependency in order.
If there is a target: check if the target fulfills the dependency requirement. If it doesn't, add/check the first dependency in order. If it does, then we don't need to add any dependency, just install the target since it already targeting a dependency.   There might be some additional logic modification downstream where if install failed from the target dependency, we want to not force install the other dependency, and fail to install the OHI.

Reply:
To discuss, a new PR: https://github.com/newrelic/newrelic-cli/pull/1536